### DISCUSSION TITLE
Is there a way to make Image Style Filter use the GPU for computation?

### DISCUSSION THREAD

**Discussion by molkemon** [WASasquatch/was-node-suite-comfyui#231] [General]
Hello, currently the image style filter is CPU-only, this is clearly visible from watching task manager. 

I am trying out a comfy workflow that does not use any AI models, just controlnet preprocessors and image blending/sharpening, and then an Image style filter. 

As far as I can tell none of the nodes use the GPU for computation, but the Image Style Filter takes the longest by far, almost 3 seconds per frame which is too long for what I want it to do. 

Is there a way to make it use GPU accelleration, or better yet, make all nodes (like blend/sharpen etc) use GPU acceleration?

**Comment by WASasquatch** [comment]
Image Style Filter is based on Pilgram package (which is old) which mimics old Instagram filters. I don't have plans to re-implement all those procedures on torch. What filter are you using? You may be able to mimic it with gradient mapping or other nodes. 

FYI that's one of the first nodes I made, and haven't even looked at how Pilgram is reproducing those old isntgram filters.
